pkg/ingress/query_test.go

```diff
@@ -287,7 +287,13 @@ func TestQueryIngressClass(t *testing.T) {
 
 	assert.NotNil(t, dep.Spec.IngressClassName)
 	assert.Equal(t, "nginx", *dep.Spec.IngressClassName)
-	assert.Nil(t, ingressNoClass.Get().Spec.IngressClassName)
+	_, err := getIngressClass()
```

Comment:
Create a separate test for this method.

Also, in this test, you should test:
1. What does it happen when you don't provide an Ingress class in a Jaeger instance and there is a default Ingress class in the cluster?
1. What does it happen when you provide an Ingress class in a Jaeger instance and there is a default Ingress class in the cluster?

Reply:
And in this do i need to add 3rd case 
3. What happen if you don't provide an Ingress class in a Jaeger instance and there is no default Ingress class in the cluster but there is nginx ingress controller available in cluster. (as in this case issue #1568)
In last one it will add nginx to Jaeger ingress. [List of available ingress controllers](https://docs.google.com/spreadsheets/d/191WWNpjJ2za6-nbG4ZoUMXMpUK8KlCIosvQB0f-oq3k/edit#gid=907731238)